### PR TITLE
Update GSoC2022 for TuringLang

### PR DESCRIPTION
**Pull request by xukai92**
TuringLang's GSoC project proposals for 2022

### CODE REVIEW

File: jsoc/gsoc/turing.md [hunk 1]
```diff
@@ -5,65 +5,106 @@
...
+**Project length**: 175 hrs
 
-Students will work with the code at [AdvancedMH](https://github.com/TuringLang/AdvancedMH.jl) or [AdvancedHMC](https://github.com/TuringLang/AdvancedHMC.jl), depending on their interests.
+**Description**: 
+Tape caching often leads to significant performance improvements for gradient-based sampling algorithms (e.g. HMC/NUTS). Tape caching is only possible at the complete computational level for ReverseDiff at the moment. This project is about implementing a more modular, i.e. function-as-a-caching-barrier, tape caching mechanism for ReverseDiff.jl.
 
-## Documentation and tutorial improvements
+## Benchmarking & improving performance of the JuliaGaussianProcesses libraries
 
 **Mentors**: Cameron Pfiffer, Martin Trapp
```
Comment (theogf) [inline jsoc/gsoc/turing.md:70]:
```suggestion
**Mentors**: Theo Galy-Fajou, Will Tebbutt, ST John
```

File: jsoc/gsoc/turing.md [hunk 2]
```diff
@@ -5,65 +5,106 @@
...
 
-**Project difficulty**: Easy
+**Project difficulty**: Medium
+
+**Project length**: 350 hrs
 
-**Description**: Turing's documentation and tutorials need a bit of an overhaul. Turing has changed significantly since the last time the documentation was written, and it's beginning to show. Students would use their knowledge of probabilistic programming languages and Turing to shore-up or rewrite documentation and tutorials.
+**Description**: 
+Turing's documentation and tutorials need a bit of an overhaul. Turing has changed significantly since the last time the documentation was written, and it's beginning to show. Students would use their knowledge of probabilistic programming languages and Turing to shore-up or rewrite documentation and tutorials.
```
Comment (theogf) [inline jsoc/gsoc/turing.md:77]:
```suggestion
Although KernelFunctions.jl has extensive correctness testing, our performance testing is lacking. This project aims to resolve this, and resolve performance issues wherever they are found. The student would first need to extend our existing benchmarking coverage, and debug and obvious performance problems. The next phase of the work would be to construct end-to-end examples of KernelFunctions being used in practice, profile them to determine where performance problems lie there, and fix them.
```

File: jsoc/gsoc/turing.md [hunk 3]
```diff
@@ -5,65 +5,106 @@
...
+**Project length**: 350 hrs
+
+**Description**:
+Adding [approximate inference](https://github.com/JuliaGaussianProcesses/JuliaGaussianProcesses.github.io/discussions/5#discussioncomment-1627101) methods for non-Gaussian likelihoods which are available in other GP packages but not yet within JuliaGPs. The project would start by determining which approximate inference method(s) to implement---there's lots to do, and we're happy to work with a student on whichever method they are most interested in, or to suggest one if they have no strong preference.
+
+## GPU integration in the JuliaGPs ecosystem
+
+**Mentors**: Ross Viljoen, Theo Galy-Fajou, Will Tebbutt
 
 **Project difficulty**: Easy to Medium
```
Comment (theogf) [inline jsoc/gsoc/turing.md:105]:
```suggestion
**Project difficulty**: Medium
```

File: jsoc/gsoc/turing.md (hunk 2 above)
Comment (st--) [reply jsoc/gsoc/turing.md:77]: Here with some minor typo editing in Theo's suggestion:
```suggestion
Although KernelFunctions.jl has extensive correctness testing, our performance testing is lacking. This project aims to resolve this, and resolve performance issues wherever they are found. The student would first need to extend our existing benchmarking coverage, and debug any obvious performance problems. The next phase of the work would be to construct end-to-end examples of KernelFunctions being used in practice, profile them to determine where performance problems lie, and fix them.```

Comment (theogf) [reply jsoc/gsoc/turing.md:77]: You have left some ``` at the end

**Comment by logankilpatrick** [comment]
Thank you for updating this!

**Pull request merged**